### PR TITLE
Move files from /pure to the /common (and other) directories (part 3 of 3)

### PR DESCRIPTION
**Pull request by robertbrignull**
Follows on from #2532 and #2530

Addresses the following:
- `legacy-messages.ts`
- `messages-shared.ts`
- `new-messages.ts`
- `log-summary-parser.ts`
- `result-keys.ts`
- `variant-analysis-filter-sort.ts`

Each file is handled in a separate commit.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by charisk — APPROVED** [review]

**Comment by robertbrignull** [comment]
> The "Move new-messages.ts to /query-server" commit shows a deletion of the new-messages.ts file, and "Move log-summary-parser.ts to /log-insights" shows it being added again.

Oops I may have messed up the commits by forgetting to add new files or accidentally changing something while the commit linting is still happening 🤦🏼 

> Will you be updating the test paths in a follow-up PR? I noticed that the unit-tests/pure directory still contains all files which have been moved to common.

That's a very good point. I'd forgotten about the tests. I don't think it's worth trying to add them all into this PR since I've already done two other PRs that omitted tests too. I'll open another PR after this one to update all the tests as well.

**Comment by robertbrignull** [comment]
I've fixed the commits so everything happens in the right one now.

File: extensions/ql-vscode/src/variant-analysis/variant-analysis-filter-sort.ts [hunk 1]
```diff
@@ -1,7 +1,4 @@
-import {
-  Repository,
-  RepositoryWithMetadata,
-} from "../variant-analysis/shared/repository";
+import { Repository, RepositoryWithMetadata } from "./shared/repository";
```
[the remark this answers is not in the document]
Comment (robertbrignull) [reply extensions/ql-vscode/src/variant-analysis/variant-analysis-filter-sort.ts]: Makes sense to me. I'll move it there.

I wondered what we had already for code shared between the host and view.

**Review by koesie10 — APPROVED** [review]
Thanks for making those changes!

**Pull request merged**